commit 8e8560a6d914929ab059233a6ecdc19e6898f299
Author: Phil Blundell <philb@gnu.org>
Date:   Wed Nov 24 11:49:53 2010 -0800

    econet: fix CVE-2010-3850
    
    Add missing check for capable(CAP_NET_ADMIN) in SIOCSIFADDR operation.
    
    Signed-off-by: Phil Blundell <philb@gnu.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/net/econet/af_econet.c b/net/econet/af_econet.c
index e622331..1d96608 100644
--- a/net/econet/af_econet.c
+++ b/net/econet/af_econet.c
@@ -663,6 +663,9 @@ static int ec_dev_ioctl(struct socket *sock, unsigned int cmd, void __user *arg)
 	err = 0;
 	switch (cmd) {
 	case SIOCSIFADDR:
+		if (!capable(CAP_NET_ADMIN))
+			return -EPERM;
+
 		edev = dev->ec_ptr;
 		if (edev == NULL) {
 			/* Magic up a new one. */
